### ISSUE TITLE
nice-grpc-web fetch client does not cache CORS responses in chromium browsers

### ISSUE THREAD

**Issue by davidfiala** [deeplay-io/nice-grpc#349]
The FetchTransport which I believe was [introduced here](https://github.com/deeplay-io/nice-grpc/pull/274/files#diff-0f02e700c1acad66a73b6ecfaad96d57679c7b7425297fd8522a07b6d6048e31R49) adds a `cache: 'no-cache'` param to the `fetch` API.

Present here is the `no-cache` param being added to the `FetchTransport`: https://github.com/deeplay-io/nice-grpc/blob/4edf6a145cefe65f23795e53dc300d681095c6ec/packages/nice-grpc-web/src/client/transports/fetch.ts#L56

✅ In firefox & safari, when no-cache is specified, the CORS OPTIONS request remains cached.
❌ However, in Chromium browsers they always perform a CORS OPTIONS request again. (Chrome, Edge)

(The above statements assume that a proper `Access-Control-Max-Age` was present to be cached)

I'm, not sure what the correct behavior is within the spec, however the discrepancy does cause serious performance issues regarding latency.

I'd like to propose adding an optional `cache?: RequestCache` field to the FetchTransportConfig:

```
export interface FetchTransportConfig {
  credentials?: RequestCredentials;
  cache?: RequestCache; // default to 'no-cache' if left undefined
}
```

and

```
    const response = await fetch(url, {
      method: 'POST',
      body: requestBody,
      headers: metadataToHeaders(metadata),
      signal,
      cache: config?.cache ? config.cache : 'no-cache',
      ['duplex' as any]: 'half',
      credentials: config?.credentials,
    });
```

I think the area for discussion here is whether to retain the default as 'no-cache', or allow undefined values which browsers would interpret as 'default'. Link to possible values here: https://javascript.info/fetch-api#cache

Arguably no-cache is safer, but will provide surprising latency in Chromium browsers and catch unsuspecting developers off guard.

**Comment by aikoven** [comment]
Thank you for your report!

I'm actually not sure if we should even set `no-cache` by default. I can't remember exactly why it was set.

What do you think about removing the default, but still letting users to specify it if needed?

**Comment by aikoven** [comment]
Thanks! I'll make a release shortly.

**Comment by aikoven** [comment]
Released in 📦 `nice-grpc-web@3.3.0`.
